This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kleis Auke Wolthuizen <github@kleisauke.nl>
Date: Mon, 25 Nov 2019 15:30:00 +0100
Subject: [PATCH 1/3] Remove -D_FORTIFY_SOURCE=2 flag

See:
https://github.com/msys2/MINGW-packages/issues/5803
https://sourceforge.net/p/mingw-w64/bugs/818/

diff --git a/build/configure.ac.warnings b/build/configure.ac.warnings
index 1111111..2222222 100644
--- a/build/configure.ac.warnings
+++ b/build/configure.ac.warnings
@@ -38,9 +38,6 @@ dnl We also abuse the warning-flag facility to enable other compiler
 dnl options.  Namely, the following:
 MAYBE_WARN="$MAYBE_WARN -fno-strict-aliasing -fno-common"
 
-dnl Also to turn various gcc/glibc-specific preprocessor checks
-MAYBE_WARN="$MAYBE_WARN -Wp,-D_FORTIFY_SOURCE=2"
-
 # invalidate cached value if MAYBE_WARN has changed
 if test "x$cairo_cv_warn_maybe" != "x$MAYBE_WARN"; then
 	unset cairo_cv_warn_cflags

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kleis Auke Wolthuizen <github@kleisauke.nl>
Date: Wed, 12 Feb 2020 20:00:00 +0100
Subject: [PATCH 2/3] Always define _cairo_utf8_to_utf16 on Windows

_cairo_utf8_to_utf16 is used by _cairo_fopen on Windows so it must
be defined even if Win32 fonts or other features are disabled.

diff --git a/src/cairoint.h b/src/cairoint.h
index 1111111..2222222 100644
--- a/src/cairoint.h
+++ b/src/cairoint.h
@@ -1913,7 +1913,7 @@ cairo_private int
 _cairo_ucs4_to_utf16 (uint32_t    unicode,
 		      uint16_t   *utf16);
 
-#if CAIRO_HAS_WIN32_FONT || CAIRO_HAS_QUARTZ_FONT || CAIRO_HAS_PDF_OPERATORS
+#if _WIN32 || CAIRO_HAS_WIN32_FONT || CAIRO_HAS_QUARTZ_FONT || CAIRO_HAS_PDF_OPERATORS
 # define CAIRO_HAS_UTF8_TO_UTF16 1
 #endif
 #if CAIRO_HAS_UTF8_TO_UTF16

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kleis Auke Wolthuizen <github@kleisauke.nl>
Date: Fri, 20 Mar 2020 13:00:00 +0100
Subject: [PATCH 3/3] Avoid mapping platform-specific functions on MinGW


diff --git a/src/cairo-compiler-private.h b/src/cairo-compiler-private.h
index 1111111..2222222 100644
--- a/src/cairo-compiler-private.h
+++ b/src/cairo-compiler-private.h
@@ -195,7 +195,7 @@
 #define __attribute__(x)
 #endif
 
-#if (defined(__WIN32__) && !defined(__WINE__)) || defined(_MSC_VER)
+#if (defined(__WIN32__) && !defined(__WINE__) && !defined(__MINGW32__)) || defined(_MSC_VER)
 #define access _access
 #define fdopen _fdopen
 #define hypot _hypot
